src/components/Dialog.vue

```diff
@@ -0,0 +1,160 @@
+    >
+      <div class="dialog">
+        <wra-header size="h3" id="dialog-title">
+          {{ title }}
+        </wra-header>
+        <p id="dialog-message">{{ message }}</p>
+        <div class="dialog-buttons">
+          <wra-button
+            ref="confirmBtn"
+            :backgroundColor="isDelete ? 'wra-red' : 'wra-green'"
+            @click="confirm"
+          >
+            {{ confirmText }}
+          </wra-button>
+          <wra-button ref="cancelBtn" @click="cancel">
+            {{ cancelText }}
+          </wra-button>
+        </div>
+      </div>
+    </div>
+  </teleport>
+</template>
+
+<script>
+import WraHeader from "./Header.vue";
+import WraButton from "./Button.vue";
+
+export default {
+  name: "wra-dialog",
+  components: {
+    WraHeader,
+    WraButton
+  },
+  props: {
+    visible: {
+      type: Boolean,
+      required: true
+    },
+    title: {
+      type: String,
+      required: true
+    },
+    message: {
+      type: String,
+      required: true
+    },
+    confirmText: {
+      type: String,
+      required: true
+    },
+    cancelText: {
+      type: String,
+      required: true
+    },
+    isDelete: {
+      type: Boolean,
+      default: false
+    }
+  },
+  emits: ["confirm", "cancel"],
+  methods: {
+    confirm() {
+      this.$emit("confirm");
+    },
+    cancel() {
+      this.$emit("cancel");
+    },
+    handleTab(e) {
+      // Trap focus between the two buttons
+      const focusable = [
+        this.$refs.confirmBtn?.$el || this.$refs.confirmBtn,
+        this.$refs.cancelBtn?.$el || this.$refs.cancelBtn
+      ].filter(Boolean);
+      if (!focusable.length) return;
+      const index = focusable.indexOf(document.activeElement);
+      if (e.shiftKey) {
+        // Shift+Tab
+        const prev = (index - 1 + focusable.length) % focusable.length;
+        focusable[prev].focus();
+      } else {
+        // Tab
+        const next = (index + 1) % focusable.length;
+        focusable[next].focus();
+      }
+    }
+  },
+  mounted() {
+    this.lastFocused = document.activeElement;
+    this.$nextTick(() => {
+      // Focus the confirm button when dialog opens
+      const btn = this.$refs.confirmBtn?.$el || this.$refs.confirmBtn;
+      if (btn && typeof btn.focus === "function") btn.focus();
+    });
+  },
+  watch: {
+    visible(val) {
+      if (val) {
+        // Store the currently focused element when the dialog opens
+        this.lastFocused = document.activeElement;
+        this.$nextTick(() => {
+          const btn = this.$refs.confirmBtn?.$el || this.$refs.confirmBtn;
+          if (btn && typeof btn.focus === "function") btn.focus();
+        });
+      } else if (
+        this.lastFocused &&
+        typeof this.lastFocused.focus === "function"
+      ) {
+        // Restore focus only if the element is still in the document
+        if (document.contains(this.lastFocused)) {
+          this.lastFocused.focus();
+        }
+      }
```

Comment:
No need to double nest the if statements

```suggestion
      if (val) {
        // Store the currently focused element when the dialog opens
        this.lastFocused = document.activeElement;
        this.$nextTick(() => {
          const btn = this.$refs.confirmBtn?.$el || this.$refs.confirmBtn;
          if (btn && typeof btn.focus === "function") btn.focus();
        });
      } else if (
        this.lastFocused &&
        typeof this.lastFocused.focus === "function" &&
        document.contains(this.lastFocused)
      ) {
        // Restore focus only if the element is still in the document
        this.lastFocused.focus();
      }
```

Reply:
Completed